packages/desktop-client/src/components/manager/ConfigServer.js

```diff
@@ -12,30 +12,24 @@ import {
 } from 'loot-design/src/components/common';
 import { colors } from 'loot-design/src/style';
 
+import { useServerURL } from '../../hooks/useServerURL';
 import { Title, Input } from './subscribe/common';
 
 export default function ConfigServer() {
   let dispatch = useDispatch();
   let history = useHistory();
   let [url, setUrl] = useState('');
+  let currentUrl = useServerURL();
+  useEffect(() => {
+    setUrl(currentUrl);
```

Comment:
nitpick: race condition. What if `useServerURL` is a very slow operation? Customer is able to start typing in his custom URL and then.. boom.. the input gets reset as soon as `currentUrl` is populated.


```suggestion
    setUrl((url) => url ? url : currentUrl);
```

Reply:
I’m pretty sure `useServerURL` will be quite fast since it’s just reaching out to a local web worker, and the [corresponding backend function](https://github.com/actualbudget/actual/blob/9c0df36e16d031501474bc3ee738af24479a02a1/packages/loot-core/src/server/main.js#L1529) finishes synchronously. There could really only be delay if the system as a whole is under severe load, which would likely also make it difficult to type.

Reply:
Never overestimate network connectivity. It's fragile in most parts of the world :) 

But yea.. this is mostly a nitpick with a subtle bug that will not appear for 99% of users.

Reply:
This isn’t going over the network tho (I have lots of experience with terrible internet 😅). It just calls [`postMessage`](https://github.com/actualbudget/actual/blob/12289792dab67a0fdd99b4bb303ef89a9393f928/packages/loot-core/src/platform/client/fetch/index.browser.js#L142-L158).

Reply:
Yea. So `postMessage` sends a message to the web worker. Then the web-worker picks up this task in async and makes a HTTP fetch request (network call) to retrieve the data. This data is then returned via the [`replyHandlers`](https://github.com/actualbudget/actual/blob/12289792dab67a0fdd99b4bb303ef89a9393f928/packages/loot-core/src/platform/client/fetch/index.browser.js#L141).

When running locally, the server is also local and hence the call _might_ be instantaneous. But we must keep in mind that Actual can also be deployed to a server.

Or am I misreading something here?

Reply:
My bad: it's actually currently not going over network. I was looking at `get-server-versions`. The handler just returns constants. Though always better to assume this will change eventually to actually go over the network. Reliability through resilience :)

https://github.com/MatissJanis/actual/blob/master/packages/loot-core/src/server/main.js#L1528-L1530

Reply:
As far as I can tell this is calling out to the “server” inside of the actual client which seems to handle the business logic (while the client client is presumably focused on displaying the info). It wouldn’t make sense to ask a remote server for its URL since the app would always need to store that information locally, which suggests to me that the worker being called out to is actually local.